lib/plugins/physics.js

```diff
@@ -51,25 +54,42 @@ function inject (bot, { physicsEnabled }) {
 
   // This function should be executed each tick (every 0.05 seconds)
   // How it works: https://gafferongames.com/post/fix_your_timestep/
+
+  // WARNING: THIS IS NOT ACCURATE ON WINDOWS (15.6 Timer Resolution)
+  // use WSL or switch to Linux
+  // see: https://discord.com/channels/413438066984747026/519952494768685086/901948718255833158
   let timeAccumulator = 0
   function doPhysics () {
     const now = performance.now()
-    const deltaSeconds = (now - lastPhysicsFrameTime) / 1000
+    let deltaSeconds = (now - lastPhysicsFrameTime) / 1000
     lastPhysicsFrameTime = now
 
+    if (deltaSeconds >= PHYSICS_CATCHUP_TICKS) deltaSeconds = PHYSICS_CATCHUP_TICKS // limiter (limit catchups)
     timeAccumulator += deltaSeconds
 
     while (timeAccumulator >= PHYSICS_TIMESTEP) {
-      if (bot.physicsEnabled && shouldUsePhysics) {
-        physics.simulatePlayer(new PlayerState(bot, controlState), world).apply(bot)
-        bot.emit('physicsTick')
-        bot.emit('physicTick') // Deprecated, only exists to support old plugins. May be removed in the future
-      }
-      updatePosition(PHYSICS_TIMESTEP)
+      tickPhysics(now)
       timeAccumulator -= PHYSICS_TIMESTEP
     }
   }
 
+  let deprecatedEvent = false // remove when physicTick is removed
+  function tickPhysics (now) {
+    if (bot.blockAt(bot.entity.position) == null) return // check if chunk is unloaded
+    if (bot.physicsEnabled && shouldUsePhysics) {
+      physics.simulatePlayer(new PlayerState(bot, controlState), world).apply(bot)
+      bot.emit('physicsTick')
+      bot.emit('physicTick') // Deprecated, only exists to support old plugins. May be removed in the future
+      if (bot.listenerCount('physicTick') > 0 && deprecatedEvent === false) { // remove when physicTick is removed
+        console.log('physicTick event is deprecated! Please use physicsTick instead')
+        deprecatedEvent = true
+      }
```

Comment:
This is a hot loop, it's probably not the right place for such a check, a better place would be https://nodejs.org/api/events.html#event-newlistener

That being said, it's maybe time to remove the deprecated event...

Reply:
I could remove the deprecated event if you would like.